common/metrics/metricstest/metricstest.go

```diff
@@ -193,10 +244,52 @@ func (s Snapshot) MustGauge(name string, tags ...metrics.Tag) float64 {
 	return v
 }
 
+func (s Snapshot) Histogram(name string, tags ...metrics.Tag) ([]HistogramBucket, error) {
+	labelValues := map[string]string{}
+	for _, tag := range tags {
+		labelValues[tag.Key()] = tag.Value()
+	}
+
+	sample, ok := s.histogramSamples[name]
+	if !ok {
+		return nil, fmt.Errorf("metric %s not found", name)
+	}
+	if sample.metricType != dto.MetricType_HISTOGRAM {
+		return nil, fmt.Errorf("metric %s not a %s type", name, dto.MetricType_HISTOGRAM.String())
+	}
```

Comment:
I think we can get rid of this because this is the same as the condition we check before adding the sample

Reply:
Ahh, I see. Makes sense to keep then